connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/SparkConnectPlanExecution.scala

```diff
@@ -157,6 +161,7 @@ private[execution] class SparkConnectPlanExecution(executeHolder: ExecuteHolder)
 
             // Collect errors and propagate them to the main thread.
             future.onComplete { result =>
+              executePlan.events.postFinished()
```

Comment:
in case of error, do you want to have error after finished (like you have here), or would you rather just post error and don't post finished? (as the query might have not finished before error).
Thriftserver would post error without finished i believe.

Reply:
Agreed we should send finished on success only. Moved to result.foreach